### PR TITLE
Add onTokenRefresh callback

### PR DESCRIPTION
**Pull request by Krakaw**
When a token is refreshed, call the supplied callback

### CODE REVIEW

File: src/classes/xhr-client.ts [hunk 1]
```diff
@@ -44,6 +46,10 @@ export default class XhrClient {
 	) {
 		this.server = server;
 		this.serverInterface = server.serverInterface;
+		let {
```
Comment (coka) [inline src/classes/xhr-client.ts]: This can be `const`.

File: src/classes/xhr-client.ts [hunk 2]
```diff
@@ -44,6 +46,10 @@ export default class XhrClient {
...
+			onTokenRefresh, ...restClientParams
```
Comment (coka) [inline src/classes/xhr-client.ts]: Nice way to keep it open for extension. :+1:

**Comment by coka** [comment]
@Krakaw Haha, my approving review is not approving enough as far as GitHub permission are concerned. :joy:

File: src/classes/xhr-client.ts (hunk 1 above)
Comment (Krakaw) [inline src/classes/xhr-client.ts]:
```suggestion
		const {
```

**Pull request merged**